accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,188 @@
+                <button #workspaceCopyButton id="{{treeService.createId(workspaceCopyButton)}}" (click)="copyToWorkspace(block)">copy to workspace</button>
+                {{setLabelledBy(workspaceCopy, concatStringWithSpaces(workspaceCopyButton.id, "blockly-button"))}}
+              </li>
+              <li #blockCopy id="{{treeService.createId(blockCopy)}}" role="treeitem" aria-selected=false [attr.aria-level]="level+2">
+                <button #blockCopyButton id="{{treeService.createId(blockCopyButton)}}" (click)="sharedClipboardService.copy(block)">copy to clipboard</button>
+                {{setLabelledBy(blockCopy, concatStringWithSpaces(blockCopyButton.id, "blockly-button"))}}
+              </li>
+              <li #sendToSelected id="{{treeService.createId(sendToSelected)}}" role="treeitem" aria-selected=false [attr.aria-level]="level+2">
+                <button #sendToSelectedButton id="{{treeService.createId(sendToSelectedButton)}}" (click)="copyToMarked(block)" disabled="{{markedBlockCompatibilityHTMLText(block)}}" [attr.aria-disabled]="markedBlockCompatibilityHTMLText(block)">copy to marked spot</button>
+                {{setLabelledBy(sendToSelected, concatStringWithSpaces(sendToSelectedButton.id, "blockly-button", markedBlockCompatibilityHTMLText(block)))}}
+              </li>
+            </ol>
+            {{addClass(listItem, "blocklyHasChildren")}}
+          </li>
+          <div *ngFor="#inputBlock of block.inputList; #i=index">
+            <field-view [attr.aria-level]="level+1" *ngFor="#field of getInfo(inputBlock); #j=index" [field]="field" [level]="level+1"></field-view>
+            <toolbox-tree-view *ngIf="inputBlock.connection && inputBlock.connection.targetBlock()" [block]="inputBlock.connection.targetBlock()" [displayBlockMenu]="false" [level]="level+1"></toolbox-tree-view>
+            <li aria-selected=false #listItem1 role="treeitem" [attr.aria-level]="level+1" id="{{treeService.createId(listItem1)}}" *ngIf="inputBlock.connection && !inputBlock.connection.targetBlock()">
+              <label #label id="{{treeService.createId(label)}}">{{getInputTypeLabel(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:</label>
+              {{setLabelledBy(listItem1, concatStringWithSpaces("blockly-argument-text", label.id))}}
+            </li>
+          </div>
+        </ol>
+      </li>
+      <toolbox-tree-view *ngIf= "block.nextConnection && block.nextConnection.targetBlock()" [level]="level" [block]="block.nextConnection.targetBlock()" [displayBlockMenu]="false"></toolbox-tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.ToolboxTreeView; }),
+        blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+                  function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
```

Comment:
Set this to Object.create(null), rather than {}. The former has no inherited properties that pollute your object, whereas the latter has 'toString' and a bunch of others that in theory could collide with your data.
Applies to anywhere an object is being used as a map.